lib/itax_code/parser.rb

```diff
@@ -19,7 +19,7 @@ class Parser
     InvalidTaxCodeError               = Class.new(Error)
 
     def initialize(tax_code, utils = Utils.new)
-      @tax_code = tax_code.upcase
+      @tax_code = tax_code.to_s.upcase
```

Comment:
@baraidrissa nice catch 🙌🏻  Penso sia meglio usare il safe navigator:

```suggestion
      @tax_code = tax_code&.upcase
```

Potresti aggiungere un test per coprire il caso in cui `ItaxCode::Parser` venga initializzato con `tax_code` a nil?